util/container/Dockerfile

```diff
@@ -35,18 +35,24 @@ RUN apt-get update && apt-get install -y lsb-release wget software-properties-co
   vim \
   nano \
   openjdk-11-jre-headless openjdk-11-jdk-headless \
-  mlir-18-tools clang-format-18 python3 \
+  mlir-17-tools clang-format-17 python3 \
   help2man perl make autoconf g++ flex bison ccache \
   libgoogle-perftools-dev numactl perl-doc \
   libfl2 libfl-dev zlib1g zlib1g-dev \
   sbt \
   python3-pip && \
-  # create symbolic links for all *-18 binaries
-  for f in /usr/bin/*-18; do ln -s $f ${f%-18}; done && \
+  # create symbolic links for all *-17 binaries
+  for f in /usr/bin/*-17; do ln -s $f ${f%-17}; done && \
   # Enable break-system-packages for non-venv package installs
   python3 -m pip config set global.break-system-packages true
 
 
+# Temporarily install python3.10 until xdsl has moved to newer MLIR version
+RUN add-apt-repository -y ppa:deadsnakes/ppa && \
+  apt update && \
+  apt install -y python3.10 && \
+  ln -fs /usr/bin/python3.10 /usr/bin/python3 && \
```

Comment:
This looks rather dangerous, since some packages might expect python 3.12 to work here. But if it works it works I guess?

Reply:
I haven't noticed any problems, but let's find out. This is hopefully quite temporary